sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -71,7 +72,26 @@ class QueryExecution(
 
   lazy val analyzed: LogicalPlan = executePhase(QueryPlanningTracker.ANALYSIS) {
     // We can't clone `logical` here, which will reset the `_analyzed` flag.
-    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+    val plan = sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+
+    // For various commands (like DDL) and queries with side effects, we force query execution
+    // to happen right away to let these side effects take place eagerly.
+    def runCommand(plan: LogicalPlan): LogicalPlan = {
+      val qe = new QueryExecution(sparkSession, plan, isCommand = true)
+      LocalRelation(plan.output,
+        SQLExecution.withNewExecutionId(qe)(qe.executedPlan.executeCollect()),
+        fromCommand = true)
+    }
+
+    if (isCommand) {
+      plan
+    } else {
+      plan.transformUp {
```

Comment:
the previous logic only handles top-level Command and Union of Command, why do we change to `transformUp` here?

Reply:
oops... should follow the previous behavior, will fix.